techmatters/flex-plugins
# CSAM: Update Contact Details To Show Self Reported CSAM

Primary reviewer: @stephenhand or @acedeywin

This PR depends on https://github.com/techmatters/hrm/pull/268.

## Description
Adds logic to support issuing `self-generated` CSAM reports, store them in HRM and display in Contact Summary (tabbed forms) and Contact Details.

![Screenshot from 2022-11-25 14-46-36](https://user-images.githubusercontent.com/15805319/204035699-b22f8e7d-e3df-4207-8db7-6e9fbc3e3b3e.png)
![Screenshot from 2022-11-25 14-45-57](https://user-images.githubusercontent.com/15805319/204035717-3ff9eea0-f9d4-4824-8f21-b1e0f49d74fb.png)

NOTE: 
- Instead than removing "oprhan CSAM records" in case of errors from external API, we could be add some retry logic, if considered worth it.
- This PR does not address the "report status screen", since that will depend on the external API rather than HRM.

### Checklist
- [x] [Corresponding issue has been opened](https://bugs.benetech.org/browse/CHI-1454)
- [WIP] New tests added
- [n/a] Feature flags added
- [n/a] Strings are localized
- [n/a] Tested for chat contacts
- [n/a] Tested for call contacts

### Verification steps
- Start local hrm sitting in https://github.com/techmatters/hrm/pull/268, expose the endpoing (using `ngrok`?).
- `npm run dev`, and point to the local server (in Chrome console, `Twilio.Flex.Manager.getInstance().serviceConfiguration.attributes.hrm_base_url = <the url of step 1>`).
- Save some CSAM reports of both kinds (self generated and counsellor generated).
- Confirm that reports are properly stored, and linked to the contact once this is saved.
- Force an error in line 137 of `plugin-hrm-form/src/components/CSAMReport/CSAMReport.tsx` and confirm that the orphan report is removed from the DB if it can't be aknowledged.
